data/libs/SpaceStation.lua

```diff
@@ -60,9 +70,9 @@ local function updateEquipmentStock (station)
 				elseif pricemod > 4 then --minor import
 					stock = stock - (rn*0.07)     -- shifting .07 = 1% chance of 0 stock
 				elseif pricemod < -10 then --major export
-					stock = stock + (rn*0.8)
+					stock = stock +(recentSystem and rn or (rn * 0.6))	--decrease stock if player is 6 game months old customer
```

Comment:
Ugh, I hate Lua "ternaries".

`recentSystem and rn or (rn * 0.6)` is `(recentSystem and rn) or (rn * 0.6)`. So if `recentSystem` isn't nil, we get `rn`, otherwise we get `rn * 0.6`. But that assuming `rn` will always be truthy, otherwise...

Are those a regular occurrence in the Pioneer codebase nowadays?

Reply:
I surely copied from somewhere in code. Now you say, it must be from some old code. I definitely prefer yours as cleaner, without the presumption that lua will interpret 'and' before 'or'. At the moment, i did not want to diverge from what i saw in other places. Thank you for the detail.

Reply:
As long as `rn` is neither `nil` nor `false`, lua ternaries are fine. Yeah, they're kind of a semantic pain when you have to deal with falsy values but I don't consider them something worth rewriting given that the other alternative is usually 3-4 lines.